### PR TITLE
Enhancements to module:status command

### PR DESCRIPTION
**Pull request by jissereitsma**
### Description
The CLI command `module:status` provides a listing of all enabled and disabled modules. However, it is hard to determine whether a certain module is enabled or disabled: You always need scrolling and the output is not easily parsed for further automation. This PR adds a couple of things:

First of all, you can inspect the status of a specific module using a command `module:status Foo_Bar` which either reports `Module is enabled` or `Module is disabled` or a warning `Module does not exist`.

Second, you can add `--enabled` or `--disabled` to only list either enabled modules or disabled modules. I didn't make this bastard-proof, so specifying both options at once (`--enabled --disabled`) will just run (the `enabled` flag is interpreted first in PHP.

The original behaviour of `bin/magento module:status` remains untouched. The entire CLI class is cleaned up: Importing namespaces, PHP 7 argument hinting and method extraction.

### Fixed Issues (if relevant)
Not relevant.

### Manual testing scenarios
With this PR you are able to run the commands as described above.

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

File: setup/src/Magento/Setup/Console/Command/ModuleStatusCommand.php [hunk 1]
```diff
@@ -47,23 +56,101 @@ protected function configure()
...
+    /**
+     * @param string $moduleName
+     * @param OutputInterface $output
+     */
+    private function showSpecificModule(string $moduleName, OutputInterface $output)
+    {
+        $allModules = $this->getAllModules();
+        if (!in_array($moduleName, $allModules->getNames())) {
+            $output->writeln('<error>Module does not exist</error>');
+            return;
```
Comment (fooman) [inline setup/src/Magento/Setup/Console/Command/ModuleStatusCommand.php]: Maybe we can add `return \Magento\Framework\Console\Cli::RETURN_FAILURE;` here to indicate something went wrong similar to here 
https://github.com/jissereitsma/magento2/blob/047097beeed4d70f437270db34681f9022ffee67/setup/src/Magento/Setup/Console/Command/AbstractMaintenanceCommand.php#L86

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=15543) <br/>All committers have signed the CLA.

File: setup/src/Magento/Setup/Console/Command/ModuleStatusCommand.php (hunk 1 above)
Comment (jissereitsma) [reply setup/src/Magento/Setup/Console/Command/ModuleStatusCommand.php]: Good point. I was in a rush to finalize things, but did not like the return of no error nice anyway. This is fixed now in https://github.com/magento/magento2/pull/15543/commits/ca4282e5c95dd1436231aaa3f7c94995231d27a8

**Review by vadimjustus — APPROVED** [review]
[x] Review looks good
[x] Local testing passed

**Comment by magento-engcom-team** [comment]
Hi @vadimjustus, thank you for the review.
[ENGCOM-2162](https://jira.corp.magento.com/browse/ENGCOM-2162) has been created to process this Pull Request

**Comment by magento-engcom-team** [comment]
Hi @jissereitsma. Thank you for your contribution. 
We will aim to release these changes as part of 2.2.6. 
Please check the release notes for final confirmation.

Please, consider to port this solution to 2.3 release line.
You may use [Porting tool](https://github.com/magento/magento2/wiki/Simple-way-to-port-your-solutions-automatically) to port commits automatically.

**Comment by jissereitsma** [comment]
Awesome thanks. I'll work on Magento 2.3 port soon.

**Pull request merged**